src/puppetlabs/jdbc_util/core.clj

```diff
@@ -331,11 +331,17 @@
 
 (defn obj->jsonb
   "Builds a JSONB object from a given object. All insertions into JSONB
-  columns must be wrapped with this function."
-  [value]
-  (doto (PGobject.)
-    (.setType "jsonb")
-    (.setValue (json/generate-string value))))
+  columns must be wrapped with this function. Returns nil if the value is nil,
+  to avoid writing 'null'::jsonb and potentially bypassing NOT NULL
+  constraints. This behavior can be overridden by passing `true` for the
+  `allow-null?` argument."
+  ([value]
+   (obj->jsonb value false))
+  ([value allow-null?]
```

Comment:
I feel like an options map makes the calling code much more readable, but constructing a map does have a small overhead.